File: invoice-manager-services/src/main/java/com/business/unknow/services/entities/Cfdi.java
package com.business.unknow.services.entities;

import java.math.BigDecimal;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EntityListeners;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Builder(toBuilder = true)
@AllArgsConstructor(access = AccessLevel.PACKAGE)
@NoArgsConstructor(access = AccessLevel.PACKAGE)
@Setter
@Getter
@ToString
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "CFDI")
public class Cfdi {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "ID_CFDI")
  private Integer id;

  @Column(name = "VERSION")
  private String version;

  @Column(name = "SERIE")
  private String serie;

  @Column(name = "FOLIO")
  private String folio;

  @Column(name = "SELLO")
  private String sello;

  @Column(name = "NO_CERTIFICADO")
  private String noCertificado;

  @Column(name = "CERTIFICADO")
  private String certificado;

  @Column(name = "MONEDA")
  private String moneda;

  @Column(name = "TIPO_CAMBIO")
  private BigDecimal tipoCambio;

  @Column(name = "IMP_TRASLADADOS")
  private BigDecimal impuestosTrasladados;

  @Column(name = "IMP_RETENIDOS")
  private BigDecimal impuestosRetenidos;

  @Column(name = "SUB_TOTAL")
  private BigDecimal subtotal;

  @Column(name = "DESCUENTO")
  private BigDecimal descuento;

  @Column(name = "TOTAL")
  private BigDecimal total;

  @Column(name = "TIPO_COMPROBANTE")
  private String tipoDeComprobante;

  @Column(name = "METODO_PAGO")
  private String metodoPago;

  @Column(name = "FORMA_PAGO")
  private String formaPago;

  @Column(name = "CONDICIONES_PAGO")
  private String condicionesDePago;

  @Column(name = "LUGAR_EXPEDICION")
  private String lugarExpedicion;
}

File: invoice-manager-ui/src/app/@core/models/cfdi/cfdi.ts
import { CfdiRelacionados } from './cfdi-relacionados';
import { Concepto } from './concepto';
import { Emisor } from './emisor';
import { Impuesto } from './impuesto';
import { InformacionGlobal } from './informacion-global';
import { Receptor } from './receptor';

export class Cfdi {
    public version: string;
    public serie: string;
    public folio: string;
    public fecha: string;
    public sello: string;
    public formaPago: string;
    public metodoPago: string;
    public noCertificado: string;
    public certificado: string;
    public condicionesDePago: string;
    public subtotal: number;
    public descuento: number;
    public total: number;
    public moneda: string;
    public tipoCambio: number;
    public tipoDeComprobante: string;
    public exportacion: string;
    public lugarExpedicion: string;
    public confirmacion: string;

    public informacionGlobal: InformacionGlobal;
    public cfdiRelacionados: CfdiRelacionados;
    public emisor: Emisor;
    public receptor: Receptor;

    public conceptos: Concepto[];
    public impuestos: Impuesto[];
    public complemento: any[];
    public addenda: any[];

    constructor() {
        this.version = '4.0';
        this.tipoDeComprobante = 'I';
        this.conceptos = [];
        this.impuestos = [];
        this.complemento = [];
        this.receptor = new Receptor();
        this.emisor = new Emisor();
        this.total = 0;
        this.subtotal = 0;
        this.descuento = 0;
        this.tipoCambio = 1.0;
        this.metodoPago = 'PUE';
        this.formaPago = '99';
        this.moneda = 'MXN';
        this.exportacion = '01';
    }
}
